embasi/embedding.py

```diff
@@ -906,3 +919,173 @@ def run(self):
+
+        self.calc_names = ["MU0","F2A1","F1A2"]
+
+        super(FrozenDensityEmbedding, self).__init__(atoms,
+                                                     embed_mask,
+                                                     calc_base_ll,
+                                                     calc_base_hl,
+                                                     run_dir=run_dir)
+
+        initial_calculator    = deepcopy(self.calculator_ll)
+        low_level_calculator  = deepcopy(self.calculator_ll)
+        high_level_calculator = deepcopy(self.calculator_hl)
+
+        self.run_dir =  run_dir
+
+        initial_calculator.parameters['qm_embedding_type'] = 'frozendensity  write'
+        low_level_calculator.parameters['qm_embedding_type'] = 'frozendensity  read_and_write'
+        high_level_calculator.parameters['qm_embedding_type'] = 'frozendensity  read_and_write'
+
+        initial_calculator.parameters["aims_output"] = "rho_and_derivs_on_grid"
+        low_level_calculator.parameters['aims_output'] = "rho_and_derivs_on_grid"
+        high_level_calculator.parameters['aims_output'] = "rho_and_derivs_on_grid"
+
+        self.set_layer(atoms, "MU0", initial_calculator, 
+                       embed_mask, ghosts=2, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.set_layer(atoms, "F2A1", low_level_calculator, 
+                       embed_mask, ghosts=2, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.set_layer(atoms, "F1A2", high_level_calculator,
+                       embed_mask, ghosts=1, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.rank = MPI.COMM_WORLD.Get_rank()
+        self.ntasks = MPI.COMM_WORLD.Get_size()
+
+    def get_embedding_pot(self, atomsembed, n_scf, ):
+        return 
+
+    def run(self):
+        """ Summary 
+        """
+        import numpy as np
+
+        root_print("Embedding calculation begun...")
+
+        start = time.time()
+
+        ediff = 1.0
+        etot_prev = 0.0
+        max_cycle = 50
+        n_cycle = 0
+
+        cwd = os.path.join(os.getcwd(),self.run_dir[2:])
+    
+        MU0_opot  = os.path.join(cwd, "MU0/output_potential_00000000")
+        F2A1_opot = os.path.join(cwd, "F2A1/output_potential_00000000")
+        F1A2_opot = os.path.join(cwd, "F1A2/output_potential_00000000")
+
+        F2A1_npot = os.path.join(cwd, "F2A1/output_potential_00000000_new")
+        F1A2_npot = os.path.join(cwd, "F1A2/output_potential_00000000_new")
+
+        MU0_oden  = os.path.join(cwd, "MU0/output_density_00000000")
+        F2A1_oden = os.path.join(cwd, "F2A1/output_density_00000000")
+        F1A2_oden = os.path.join(cwd, "F1A2/output_density_00000000")
+
+        F2A1_nden = os.path.join(cwd, "F2A1/output_density_00000000_new")
+        F1A2_nden = os.path.join(cwd, "F1A2/output_density_00000000_new")
+
+        # Outer SCF loop
+        fdet_file = open("fdet.txt", "w")
```

Comment:
Would it be better to pipe to the user defined output (i.e., just normal print statements)? The MPI print infrastructure is in place to avoid any multiple printing issues. if we need to go in this direction, perhaps adding the absolute path here would be good - at the moment, the output file could end up in some unexpected place.